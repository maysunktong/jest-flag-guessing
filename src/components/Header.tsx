import Subtitle from "./Subtitle";

const Header = () => {
  return (
    <header className="text-center">
      <h1 className="font-bold uppercase text-5xl p-4 text-transparent bg-clip-text bg-gradient-to-r from-red-500 via-yellow-500 to-blue-500">
        Flags of the world
      </h1>
      <Subtitle text="beautiful" />
    </header>
  );
};
export default Header;
