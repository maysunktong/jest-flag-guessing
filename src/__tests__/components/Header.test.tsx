import { render, screen } from "@testing-library/react";
import Header from "@/components/Header";

describe("Test Header component", () => {
  test("Page title text shown", () => {
    render(<Header />);

    const pageTitle = screen.getByText(/flags of the world/i);
    expect(pageTitle).toBeInTheDocument();
  });

  test("Page title h1", () => {
    render(<Header />);

    const pageTitleByRole = screen.getByRole("heading", {
      level: 1,
      name: /flags of the world/i,
    });
    expect(pageTitleByRole).toBeInTheDocument();
  });

  test("Subtitle inside Header", () => {
    render(<Header />);

    const subTitle = screen.getByRole("heading", {
      level: 2,
      name: /our world is full of beautiful flags/i,
    });
    expect(subTitle).toBeInTheDocument();
  });

  /* Snapshot testing */
  test("Header container", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

   test("Header matches snapshot", () => {
    const { getByText } = render(<Header />);
    const heading = getByText(/flags of the world/i);
    expect(heading).toMatchSnapshot();
  });

  test('Header fragment matches snapshot', () => {
  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});
});
