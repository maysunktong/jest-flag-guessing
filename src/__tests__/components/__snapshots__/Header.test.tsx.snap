// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test Header component Header container 1`] = `
<div>
  <header
    class="text-center"
  >
    <h1
      class="font-bold uppercase text-5xl p-4 text-transparent bg-clip-text bg-gradient-to-r from-red-500 to-yellow-500"
    >
      Flags of the world
    </h1>
    <h2
      class="py-4 text-2xl"
    >
      Our world is full of 
      beautiful
       flags.
    </h2>
  </header>
</div>
`;

exports[`Test Header component Header fragment matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="text-center"
  >
    <h1
      class="font-bold uppercase text-5xl p-4 text-transparent bg-clip-text bg-gradient-to-r from-red-500 to-yellow-500"
    >
      Flags of the world
    </h1>
    <h2
      class="py-4 text-2xl"
    >
      Our world is full of beautiful flags.
    </h2>
  </header>
</DocumentFragment>
`;

exports[`Test Header component Header matches snapshot 1`] = `
<h1
  class="font-bold uppercase text-5xl p-4 text-transparent bg-clip-text bg-gradient-to-r from-red-500 to-yellow-500"
>
  Flags of the world
</h1>
`;
